commit 6f3ad4ba479157624b82b28ff12ef2d9af6182c8 (refs/changes/03/4003/1)
Author: Mike Gerdts <mike.gerdts@joyent.com>
Date:   2018-05-24T23:00:54+00:00 (1 year, 5 months ago)
    
    OS-6796 vmm_zsd_lock not held while element removed from vmm_zsd_list

diff --git a/usr/src/uts/i86pc/io/vmm/vmm_zsd.c b/usr/src/uts/i86pc/io/vmm/vmm_zsd.c
index 586773ac8e..99a581a9f1 100644
--- a/usr/src/uts/i86pc/io/vmm/vmm_zsd.c
+++ b/usr/src/uts/i86pc/io/vmm/vmm_zsd.c
@@ -171,7 +171,9 @@ vmm_zsd_destroy(zoneid_t zid, void *data)
 	mutex_exit(&zsd->vz_lock);
 	mutex_destroy(&zsd->vz_lock);
 
+	mutex_enter(&vmm_zsd_lock);
 	list_remove(&vmm_zsd_list, zsd);
+	mutex_exit(&vmm_zsd_lock);
 	kmem_free(zsd, sizeof (*zsd));
 }
 
